clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1396,42 +1396,6 @@ void KernelObjVisitor::visitArray(const CXXRecordDecl *Owner, FieldDecl *Field,
 class SyclKernelFieldChecker : public SyclKernelFieldHandler {
   bool IsInvalid = false;
   DiagnosticsEngine &Diag;
-  // Check whether the object should be disallowed from being copied to kernel.
-  // Return true if not copyable, false if copyable.
-  bool checkNotCopyableToKernel(const FieldDecl *FD, QualType FieldTy) {
-    if (FieldTy->isArrayType()) {
-      if (const auto *CAT =
-              SemaRef.getASTContext().getAsConstantArrayType(FieldTy)) {
-        QualType ET = CAT->getElementType();
-        return checkNotCopyableToKernel(FD, ET);
-      }
-      return Diag.Report(FD->getLocation(),
-                         diag::err_sycl_non_constant_array_type)
-             << FieldTy;
-    }
-
-    if (SemaRef.getASTContext().getLangOpts().SYCLStdLayoutKernelParams)
```

Comment:
Hmm... I wonder if this option needs to go away now.  It doesn't do anything without this check I think?

Reply:
Good call, I'll remove the option.